src/main/permissionsManager.ts

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {
+    PermissionRequestHandlerHandlerDetails,
+    WebContents,
+    app,
+    dialog,
+    ipcMain,
+} from 'electron';
+
+import {UPDATE_PATHS} from 'common/communication';
+import JsonFileManager from 'common/JsonFileManager';
+import {Logger} from 'common/log';
+import {t} from 'common/utils/util';
+import {isTrustedURL, parseURL} from 'common/utils/url';
+
+import {permissionsJson} from 'main/constants';
+import {localizeMessage} from 'main/i18nManager';
+import ViewManager from 'main/views/viewManager';
+import CallsWidgetWindow from 'main/windows/callsWidgetWindow';
+import MainWindow from 'main/windows/mainWindow';
+
+const log = new Logger('PermissionsManager');
+
+// supported permission types
+const supportedPermissionTypes = [
+    'media',
+    'geolocation',
+    'notifications',
+    'fullscreen',
+    'openExternal',
+    'clipboard-sanitized-write',
+];
+
+// permissions that require a dialog
+const authorizablePermissionTypes = [
+    'media',
+    'geolocation',
+    'notifications',
+];
+
+type Permissions = {
+    [origin: string]: {
+        [permission: string]: {
+            allowed: boolean;
+            alwaysDeny?: boolean;
+        };
+    };
+};
+
+export class PermissionsManager extends JsonFileManager<Permissions> {
+    handlePermissionRequest = async (
+        webContents: WebContents,
+        permission: string,
+        callback: (granted: boolean) => void,
+        details: PermissionRequestHandlerHandlerDetails,
+    ) => {
+        callback(await this.doPermissionRequest(
+            webContents.id,
+            permission,
+            details.securityOrigin ?? details.requestingUrl,
+        ));
+    }
+
+    doPermissionRequest = async (
+        webContentsId: number,
+        permission: string,
+        requestingURL: string,
+    ) => {
+        log.debug('doPermissionRequest', requestingURL, permission);
+
+        // is the requested permission type supported?
+        if (!supportedPermissionTypes.includes(permission)) {
+            return false;
+        }
+
+        // allow if the request is coming from the local renderer process instead of the remote one
+        const mainWindow = MainWindow.get();
+        if (mainWindow && webContentsId === mainWindow.webContents.id) {
+            return true;
+        }
```

Comment:
I see that this was in the old code too, but why don't we check the rest if the request is coming from the local renderer process?

Reply:
The local renderer process is trusted, and we control what's loaded into it. Unless the Mattermost Desktop App itself is malicious, we can assume that local code is trusted.

Reply:
Thanks for the explanation!